barretenberg/cpp/docs/src/sumcheck-outline.md

```diff
@@ -0,0 +1,438 @@
+# Sumcheck Implementation
+We implement a Zero-Knowledge Sumcheck protocol for relations of a very general form.  
+
+The implementation consists of several components. 
+- [Non-ZK Sumcheck:](#NonZKSumcheck) 
+	We sketch an implementation of the non-zero-knowledge Sumcheck, introduce the main abstractions and the components of the proof system. In [Witness Information Leakage](#NonZKSumcheckLeakage), we determine the sources allowing the verifier to learn the witness information during Sumcheck.
+	
+- [Masking Round Univariates witth Libra:](#LibraTechnique) 
+	To prevent the witness values from leaking through the coefficients of Sumcheck round univariates, we apply a technique introduced in <a href=" https://eprint.iacr.org/2019/317">Libra: Succinct Zero-Knowledge Proofs with Optimal Prover Computation</a>. 
+	Being represented in Lagrange basis, Libra masking polynomials lead to very simple formulas for contributions to Sumcheck round univariates, see [the following section](#LibraRoundUnivariates). 
+	In section [Libra Costs](#LibraCosts), we assess the overhead caused by adding the Libra technique. 
+	Although the contribution in field operations is almost negligible, it adds non-trivial expenses during the opening procedure.
+
+- [Masking Evaluations of Multilinear Witnesses:](#MaskingEvalsOfWitnesses) 
+	At the stage of proving their evaluations at the challenge point, the witness polynomials fed to Sumcheck must not reveal any private information. 
+    We use a modification of Construction 3 described in <a href=" https://eprint.iacr.org/2019/317">Libra</a>  allowing the prover to open a new multilinear polynomial derived from the witnesses by adding a product of a random scalar and a public quadratic polynomial in the same number of variables.  
+
+- [Total Costs:](#ZKCosts) The effect of adding Libra technique and masking evaluations of multilinear witnesses on is assessed, and the theoretical upper bound on prover's work is compared to the implemenation costs. 
+
+Non ZK-Sumcheck Outline {#NonZKSumcheck}
+========
+- - -
+ ### Sumcheck Relation {#SumcheckRelation}
+
+ Given multilinear polynomials \f$ P_1,\ldots, P_N \in \mathbb{F}[X_0,\ldots, X_{d-1}] \f$ and a polynomial \f$ F \f$ in \f$ N \f$ variables, we run Sumcheck over the polynomial
+ \f{align}{
+    \tilde{F}
+    (X_0,\ldots, X_{d-1}) =
+    pow_{\beta}(X_0,\ldots, X_{d-1}) \cdot F\left( P_1 (X_0,\ldots, X_{d-1}), \ldots, P_N (X_0,\ldots, X_{d-1}) \right)
+    \f}
+to establish that \f$ F(P_1(\vec \ell),\ldots, P_N(\vec \ell) ) = 0 \f$, i.e. that \f$ F \f$ is satisfied at every
+point of \f$\{0,1\}^d\f$.
```

Comment:
I would put \vec{l} \in {0,1}^d and maybe clarify that its representation over the boolean hypercube